model/task/task.go

```diff
@@ -1572,7 +1572,11 @@ func (t *Task) MarkEnd(finishTime time.Time, detail *apimodels.TaskEndDetail) er
 		}
```

Comment:
This logic actually doesn't make much sense; it should probably be "set it to the activated time or etc" if TimeTaken is going to be accurate